### ISSUE TITLE
Cannot force update branch error is back

### ISSUE THREAD

**Issue by axelrtgs** [sous-chefs/ruby_rbenv#289]
### :speaking_head: Foreword

Thank for taking the time to fill this bug report fully. Without it we may not be able to fix the bug, and the issue may be closed without resolution.

## :ghost: Brief Description

Looks like the git resources changes in chef 16 are at it again. `Cannot force update the current branch.` error is back again and this time it's using the legacy `deploy` branch so i'm not sure how to fix this. I checked on the ruby-build repo and theres no branch called deploy there. Not sure if this is a cookbook or upstream issue but it would be nice to get a proper fix for this so it doesn't happen again.

## :pancakes: Cookbook version

2.4.0 (latest)

## :woman_cook:  Chef-Infra Version

```
      System Info:
      ------------
      chef_version=16.2.73
      platform=ubuntu
      platform_version=18.04
      ruby=ruby 2.7.1p83 (2020-03-31 revision a0c7c23c9c) [x86_64-linux]
      program_name=/usr/bin/chef-client
      executable=/opt/chef-workstation/bin/chef-client
```

## :tophat: Platform details

```
      System Info:
      ------------
      chef_version=16.2.73
      platform=ubuntu
      platform_version=18.04
      ruby=ruby 2.7.1p83 (2020-03-31 revision a0c7c23c9c) [x86_64-linux]
      program_name=/usr/bin/chef-client
      executable=/opt/chef-workstation/bin/chef-client
```

## Steps To Reproduce

add a simple ruby resource to the recipe:

```
rbenv_ruby '2.6.1' do
  user 'user'
end
```
## :police_car: Expected behavior

Successful converge

## :heavy_plus_sign: Additional context

```
System Info:
      ------------
      chef_version=16.2.73
      platform=ubuntu
      platform_version=18.04
      ruby=ruby 2.7.1p83 (2020-03-31 revision a0c7c23c9c) [x86_64-linux]
      program_name=/usr/bin/chef-client
      executable=/opt/chef-workstation/bin/chef-client


    ================================================================================
    Error executing action `install` on resource 'rbenv_ruby[2.6.1]'
    ================================================================================

    Mixlib::ShellOut::ShellCommandFailed
    ------------------------------------
    rbenv_plugin[ruby-build] (/var/chef/cache/cookbooks/ruby_rbenv/resources/ruby.rb line 41) had an error: Mixlib::ShellOut::ShellCommandFailed: git[Install ruby-build plugin] (/var/chef/cache/cookbooks/ruby_rbenv/resources/plugin.rb line 32) had an error: Mixlib::ShellOut::ShellCommandFailed: Expected process to exit with [0], but received '128'
    ---- Begin output of git branch -f deploy 16421762a6a7901c1c12811940187e40f7dfc3b9 ----
    STDOUT:
    STDERR: fatal: Cannot force update the current branch.
    ---- End output of git branch -f deploy 16421762a6a7901c1c12811940187e40f7dfc3b9 ----
    Ran git branch -f deploy 16421762a6a7901c1c12811940187e40f7dfc3b9 returned 128
```

**Comment by thr3a** [comment]
This problem is still occurring in my environment.
What is the progress of fixing this issue?

**Comment by ramereth** [comment]
Unfortunately not but this would be a great candidate for Hacktoberfest 2020.

**Comment by axelrtgs** [comment]
@thr3a So the issue at heart is a bug in chef built in git resource. i can confirm that newer versions of chef-client don't exhibit this behaviour in my testing. We use 16.4.41 in our env and we do not have this issue anymore (even with the removed workaround) and the testing running on latest did not exhibit this issue either.

The solution to this is to update to a more recent chef client with the bugfix theres not much that can be done in this cookbook. removed the workaround in the cookbook since its no longer needed on more recent versions.

**Comment by ramereth** [comment]
@axelrtgs is this only affecting Versions `>= 16, < 16.4.41`? with Chef?

**Comment by axelrtgs** [comment]
i can tell you for sure its affect some versions inside that range. i know we noticed it when first trying out 16 in the lab then found out the git resource got a rewrite and then when we went to prod with it we pinned to version 16.4.41 which seems to work fine now. it may have been fixed in a slightly earlier version. I can leave a not in the readme that theres a chef bug that affected the cookbook in that range and that we have confirmed >=16.4.41 to be working properly now
